src/prefect/utilities/executors.py

```diff
@@ -12,11 +13,22 @@
 from concurrent.futures import TimeoutError as FutureTimeout
 from functools import wraps
 from logging import Logger
-from typing import TYPE_CHECKING, Any, Callable, Dict, List, Union, Sequence, Mapping
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Callable,
+    Dict,
+    List,
+    Union,
+    Sequence,
+    Mapping,
+    Iterator,
+)
 
 import prefect
 from prefect.exceptions import TaskTimeoutSignal
 from prefect.utilities.logging import get_logger
+from prefect.utilities.threaded_heartbeat import HeartbeatThread
```

Comment:
Is there a reason you put this into its own file?  Seems potentially unnecessary

Reply:
There's no deep reason at all--without knowledge about how to organize things I tried superficially mimic existing structure. The current CLI-heartbeat is in its own file so I put this one in its own too, is there another place that makes more sense?

Reply:
I was thinking alongside the subprocess implementation in `utilities/executor.py` so it's colocated with its use